Домашнее задание №2

evg226 opened the pull request.

evg226 committed:
Домашнее задание №2

SokolAlexander commented on src/App.js:
```diff
@@ -1,16 +1,42 @@
 
 function App() {
-  const messageSend = { body: "Я изучаю React", style: "reded" };
+  let currentAuthor = "Evgeny";
```
если это значение не изменяется, я бы вынес его за компонент и объявил через конст

SokolAlexander commented on src/App.js:
```diff
@@ -1,16 +1,42 @@
+  const [messageList, setMessageList] = useState([]);
+
+  useEffect(() => {
+    if (messageList.length > 0 && messageList[messageList.length - 1].author == currentAuthor) {
```
рекомендую избегать нестрогих сравнений

SokolAlexander commented on src/App.js:
```diff
@@ -1,16 +1,42 @@
+  useEffect(() => {
+    if (messageList.length > 0 && messageList[messageList.length - 1].author == currentAuthor) {
+      const robotSay = (robotText) => {
+        addNewMessage({ id: new Date(), author: "Robot", text: robotText });
```
используете функцию выше объявления, не рекомендую

SokolAlexander commented on src/App.js:
```diff
@@ -1,16 +1,42 @@
+
+  useEffect(() => {
+    if (messageList.length > 0 && messageList[messageList.length - 1].author == currentAuthor) {
+      const robotSay = (robotText) => {
```
мне кажется, эту функцию можно объявить в компоненте, а не внутри юзЭффект

SokolAlexander commented on src/App.js:
```diff
@@ -1,16 +1,42 @@
+        addNewMessage({ id: new Date(), author: "Robot", text: robotText });
+      }
+      robotSay(`${currentAuthor}, от Вас получено сообщение "${messageList[messageList.length - 1].text}"`);
+      setTimeout(() => {
```
Таймаут неплохо бы почистить на анмаунте

SokolAlexander commented on src/MessageInputForm.js:
```diff
@@ -0,0 +1,22 @@
+  };
+
+  return (
+    <form name="newMessage" onSubmit={sendNewMessage}>
```
круть

SokolAlexander reviewed (approved):
В принципе здорово) несколько комментов оставил)

evg226 committed:
Устранены замечания

evg226 committed:
Исправлено мелких ошибок по роботу и form-input

 1-блок отправки нового сообщения до второго ответа робота и пустых сообщений, 2-устранены ошибки сообщений робота по ключу

evg226 committed:
В app.js применен хук useCallback

Функции, передаваемые через props и используемые в др.хуках обернуты в useCallback
